install/0000_00_cluster-version-operator_01_clusteroperator.crd.yaml

```diff
@@ -1,531 +0,0 @@
-kind: CustomResourceDefinition
-apiVersion: apiextensions.k8s.io/v1beta1
-metadata:
-  name: clusteroperators.config.openshift.io
-spec:
-  additionalPrinterColumns:
-  - JSONPath: .status.versions[?(@.name=="operator")].version
-    description: The version the operator is at.
-    name: Version
-    type: string
-  - JSONPath: .status.conditions[?(@.type=="Available")].status
-    description: Whether the operator is running and stable.
-    name: Available
-    type: string
-  - JSONPath: .status.conditions[?(@.type=="Progressing")].status
-    description: Whether the operator is processing changes.
-    name: Progressing
-    type: string
-  - JSONPath: .status.conditions[?(@.type=="Degraded")].status
-    description: Whether the operator is degraded.
-    name: Degraded
-    type: string
-  - JSONPath: .status.conditions[?(@.type=="Available")].lastTransitionTime
-    description: The time the operator's Available status last changed.
-    name: Since
-    type: date
-  group: config.openshift.io
-  names:
-    kind: ClusterOperator
-    listKind: ClusterOperatorList
-    plural: clusteroperators
-    singular: clusteroperator
-    shortNames:
-    - co
-  scope: Cluster
-  subresources:
-    status: {}
-  version: v1
-  versions:
-  - name: v1
-    served: true
-    storage: true
-  validation:
-    openAPIV3Schema:
-      description: ClusterOperator is the Custom Resource object which holds the current
-        state of an operator. This object is used by operators to convey their state
-        to the rest of the cluster.
-      properties:
-        apiVersion:
-          description: 'APIVersion defines the versioned schema of this representation
-            of an object. Servers should convert recognized schemas to the latest
-            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
-          type: string
-        kind:
-          description: 'Kind is a string value representing the REST resource this
-            object represents. Servers may infer this from the endpoint the client
-            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
-          type: string
-        metadata:
-          description: ObjectMeta is metadata that all persisted resources must have,
```

Comment:
```console
$ git diff -M20 origin/pr/266^^..origin/pr/266
...
         metadata:
-          description: ObjectMeta is metadata that all persisted resources must have,
-            which includes all objects users must create.
-          properties:
...
```

Where did the metadata details go?  Don't we want those?

Reply:
@wking no, those were removed in https://github.com/openshift/api/pull/478 to conform to structural schema requirements

Reply:
> ... to conform to structural schema requirements

Can you link me to docs on these requirements?  I don't see anything that sounds like it [here][1].

[1]: https://github.com/kubernetes/community/blob/ecea14c0af705635a47d6dbc99dc050ed2373065/contributors/devel/sig-architecture/api-conventions.md#metadata

Reply:
Metadata is exclusively owned by the apiserver and apimachinery. A schema inside a CRD will interfere with apimachinery and we cannot guarantee that a CR from today works tomorrow.

Metadata is validated, but implicitly following the logic inside the kube-apiserver.